packages/@uppy/utils/src/fetcher.js

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable max-len */
+
+import NetworkError from './NetworkError.js'
+import ProgressTimeout from './ProgressTimeout.js'
+
+const noop = () => {}
+
+/**
+ * Get an abort signal that will be aborted when the specified Uppy file is removed or all uploads are cancelled.
+ *
+ * @param {Uppy} uppy
+ *  The Uppy instance.
+ * @param {string} [id]
+ *  The ID of the file to watch for removal.
+ * @param {AbortSignal} [additionalSignal]
+ *  An optional additional abort signal.
+ * @returns {AbortController}
+ *  The abort signal.
+ */
+export function getUppyAbortController(uppy, id, additionalSignal) {
+  const controller = new AbortController()
+
+  uppy.once('cancel-all', () => {
+    controller.abort()
+  })
+
+  if (id) {
+    uppy.on('file-removed', (file) => {
```

Comment:
won't this leak event listeners? we never `uppy.off` all of these events as far as i can see?